src/global/config.ts

```diff
@@ -17,10 +17,25 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-type AppConfig = { ARGO_DOCS_URL_ROOT: string };
+import { RegionKey, isRegionKey, regionKeys } from './types';
+
+type AppConfig = { ARGO_DOCS_URL_ROOT: string; REGION: RegionKey };
+
+const getRegion = () => {
+	const envRegionKey = process.env.NEXT_PUBLIC_REGION_KEY || '';
+	const normalizedKey = envRegionKey.toUpperCase().trim();
+
+	// is valid region?
+	if (isRegionKey(normalizedKey)) {
+		return normalizedKey;
+	} else {
+		throw Error(`Invalid region. Must be one of ${regionKeys.join(', ')}`);
+	}
+};
```

Comment:
I am curious how `NEXT_PUBLIC_REGION_KEY` will be used throughout the app. I suspect this will just be a display string, and if so we don't need to validate its content, just that it is a non-empty string. Therefore also no enum needed.

Reply:
It will be a const of some kind. real example from mocks to display a flag based on region, or a logo based on region `object_of_rdpc_flags[region]`

Reply:
so there is a need for a constant of some kind and a display string. 
I don't like depending on a string to be a constant because you end up string manipulating back and forth for casing

Reply:
Ideally we won't have to make code changes that are specific to any region, this approach bakes in knowledge of the regions and then neccesitates when we add new regions, change a display property of a region, etc.

We should strive as long as possible to keep the code region agnostic. If this can be handled completely with config that would be much easier. If we need to push back on some design ideas that will make all our lives easier long term.